lib/src/git.rs

```diff
@@ -1905,3 +2232,512 @@ pub fn parse_gitmodules(
         .collect();
     Ok(ret)
 }
+
+fn convert_git_fetch_output_to_fetch_result(
+    output: Output,
+    prunes: &mut Vec<String>,
+) -> Result<Output, GitFetchError> {
+    // None means signal termination:
+    // 128 is the base for the signal exit codes (128 + signo)
+    let code = output.status.code();
+    if code == Some(0) {
+        return Ok(output);
+    }
+
+    let lossy_err = String::from_utf8_lossy(&output.stderr).to_string();
```

Comment:
suggestion: Move `lossy_err` construction into lambda below? (Perhaps there's an ownership reason you can't?)